src/libraries/System.Diagnostics.Process/tests/ProcessThreadTests.cs

```diff
@@ -148,17 +149,22 @@ public async Task TestStartTimeProperty()
                 await Task.Factory.StartNew(() =>
                 {
                     p.Refresh();
-                    try
-                    {
-                        var newest = p.Threads.Cast<ProcessThread>().OrderBy(t => t.StartTime.ToUniversalTime()).Last();
-                        Assert.InRange(newest.StartTime.ToUniversalTime(), curTime - allowedWindow, DateTime.Now.ToUniversalTime() + allowedWindow);
-                    }
-                    catch (InvalidOperationException)
-                    {
-                        // A thread may have gone away between our getting its info and attempting to access its StartTime
-                    }
+
+                    int newThreadId = OperatingSystem.IsWindows() ? GetCurrentThreadId() : gettid();
+
+                    ProcessThread[] processThreads = p.Threads.Cast<ProcessThread>().ToArray();
+                    ProcessThread newThread = Assert.Single(processThreads, thread => thread.Id == newThreadId);
+
+                    Assert.InRange(newThread.StartTime.ToUniversalTime(), curTime - allowedWindow, DateTime.Now.ToUniversalTime() + allowedWindow);
                 }, CancellationToken.None, TaskCreationOptions.LongRunning, TaskScheduler.Default);
             }
+
+            [DllImport("libc")]
+            static extern int gettid();
+
+            [DllImport("kernel32.dll")]
+            [SuppressGCTransition]
+            static extern int GetCurrentThreadId();
```

Comment:
We already have this method for Windows? How was this able to compile?
https://github.com/dotnet/runtime/blob/3f607bd6061299a20d29b76914b4176b9d835f66/src/libraries/System.Diagnostics.Process/tests/ProcessThreadTests.Windows.cs#L55-L56